klog.go

```diff
@@ -866,7 +866,14 @@ func SetLogger(logr logr.Logger) {
 	logging.mu.Lock()
```

Comment:
Wouldn't it make more sense to accept logr.LogSink interface and construct the logr.Logger instead of accepting the struct itself. 

I ask as I have seen multiple cases of users being confused with logr.Logger now being struct. Even though this change is purely for performance, and doesn't have any impact, some users misread this change. For example https://github.com/kubernetes/klog/pull/252

My first though about this change is that logr just splits client API (logr.Logger) from implementation interface (logr.LogSInk). Why then klog and would accept the client log.Logger instead of implementation logr.LogSink. 

It might be a subtle difference in API, but accepting interface makes it more clearer that intention of using logr is the same.

Reply:
response is https://github.com/kubernetes/klog/pull/242#issuecomment-888574114